activerecord/lib/active_record/core.rb

```diff
@@ -227,6 +226,20 @@ def generated_association_methods
         end
       end
 
+      # Returns columns which shouldn't be exposed while calling #inspect.
```

Comment:
Should these methods be documented? They're not currently, because RDoc doesn't pick up [metaprogrammed methods](http://docs.seattlerb.org/rdoc/RDoc/Parser/Ruby.html#class-RDoc::Parser::Ruby-label-Metaprogrammed+Methods) inside an `included do` block, but that seems unintentional:
https://edgeapi.rubyonrails.org/classes/ActiveRecord/Core.html

If `filter_attributes` is a documented feature that users should set directly, it should have a changelog entry and be mentioned in the guides; but if it's an implementation detail and users should only ever set `filter_parameters`, then these accessors should be `:nodoc:`.

Reply:
I think `filter_attributes` should be public API. We have even added tests for that https://github.com/rails/rails/blob/master/activerecord/test/cases/filter_attributes_test.rb#L32. Yes, it makes sense to add changelog entry too.

> RDoc doesn't pick up metaprogrammed methods inside an included do block, but that seems unintentional:
https://edgeapi.rubyonrails.org/classes/ActiveRecord/Core.html

Yes, I'm seeing the same problem. At least we should remove `# :nodoc:` for `ClassMethods` module in order to see the non-nodoc methods in the api docs. Does it make sense?

Reply:
It looks like the module is `:nodoc:` intentionally: https://github.com/rails/rails/pull/30146

We could remove `:nodoc:` from the individual methods to make it clearer that it's not a mistake, but that seems unrelated to the rest of the changes here.

Reply:
Since we want to see `filter_attributes` in the api docs, I think I should remove `# :nodoc:` for `ClassMethods`  and add `# :nodoc:` to the methods in that module except `filter_attributes`/`=`.

But I'm not completly sure because maybe there is reason why we that can't remove `nodoc` for the `ClassMethods` since #30146?
/cc @kamipo

Reply:
Oh, I missed that the new methods here were in the `ClassMethods` module. Your suggestion makes sense to me, but I'd like to hear @kamipo's opinion too.